### DISCUSSION TITLE
code for finetuning 1024 model.

### DISCUSSION THREAD

**Discussion by SunzeY** [PixArt-alpha/PixArt-alpha#104] [Q&A] [answered]
I follow the instruction to finetune 512px and it works well. However, When switch to 1024px-MS version, it doesn't produce reasonable result. Is it possible to share a official implementation to finetune 1024px-MS version of your work? This will be really helpful!

**Comment by SunzeY** [comment]
This really drives me crazy! I have tried anything I can. All I do is change the config of Pokemon-example:
1. change config
```python
_base_ = ['PixArt-alpha/configs/pixart_config/PixArt_xl2_img1024_internalms.py']
...
data = dict(type='InternalDataMS'...
model = 'PixArtMS_XL_2'
...
```
I train the model and convert pixart model to diffuser exactly follow the same pipeline as 512px base fine-tuning pipeline. However, my result always looks like this:
![pokemon_6](https://github.com/PixArt-alpha/PixArt-alpha/assets/76475229/9d75b1f6-a33b-45e4-b64b-f5e7b7cc5de6)
However, 512px base model works well normally...
Is there anything important missing that makes Pixart-MS-1024 different from Pixart-512? I would really appreciate it if official fine-tuning demo on MS-1024 will be made available!

**Comment by SunzeY** [comment]
May be there is something wrong with learnable scale and ratio embedding? Should I keep them fixed during training?

**Comment by youngwanLEE** [comment]
@SunzeY Same situation; I'm waiting for the author's response, too.

**Reply by SunzeY** [reply]
I will email their author to try to get feedback from them.

**Comment by ruizhaocv** [comment, marked as answer]
I think this is caused by the wrong positional embedding. Set the lewei_scale to 2.0 in the config file and have a try!

**Reply by SunzeY** [reply]
Thank you so much, this solve my problem!

**Comment by MingzhouZhang** [comment]
In inference.py, no-ema checkpoint is loaded, while fine-tuning ema checkpoint is saved in 'state_dict_ema' key. Replace with the following code solved my problem.
`missing, unexpected = model.load_state_dict(state_dict['state_dict_ema'], strict=False)` 
https://github.com/PixArt-alpha/PixArt-alpha/blob/82c8559b7c11e2db47ab6d8fc0a5b5fa77f88f30/scripts/inference.py#L158

**Reply by SunzeY** [reply]
hi, I have a small question for this. if `state_dict['state_dict_ema']`'s weight is better. Why not their official implementation of converting `pth` into hugging face format load ema shadow but still use `state_dict['state_dict]`.

**Comment by heyalexchoi** [comment]
Does your config continue from the 1024 checkpoint? I had that issue.
